io_u.c

```diff
@@ -1439,6 +1440,18 @@ static void lat_target_success(struct thread_data *td)
 
 	td->latency_qd_low = td->latency_qd;
 
+	if (td->latency_qd + 1 == td->latency_qd_high) {
```

Comment:
This means we would never be OK in the case where the user set a latency maximum and we started out at it but it turned out to be OK?

Reply:
I should have explained this better. 
For latency_run == 0 case, the behavior is the same as existing code. 
For latency_run == 1 case, we hit this condition when latency_qd_low == latency_qd == (latency_qd_high - 1). With existing code, latency_qd won't increase even if we hit lat_target_success() multiple times. For example, latency_qd == 3 and latency_qd_high == 4. It is very likely that we will hit latency_target with some margin with qd of 3, but we will miss latency_target with qd of 4. With existing code, we will stick with qd of 3 for the rest of the run, and result latency is lower than latency_target. 
This change tries to fix this by allowing latency_qd to increase in such cases. The heuristic here is a little too naive though...